gusto/diagnostics.py

```diff
@@ -195,6 +196,67 @@ def compute(self, state):
         return self.field
 
 
+class SphericalComponent(DiagnosticField):
+
+    def __init__(self, name):
+        super().__init__()
+        self.fname = name
+
+    def setup(self, state):
+        if not self._initialised:
+            # check geometric dimension is 3D
+            if state.mesh.geometric_dimension() != 3:
+                raise ValueError('Spherical components only work in 3D!')
```

Comment:
This is a bit confusing as shallow water is 2D, although I see what you mean that the mesh is 3D.

Reply:
Hopefully that's better!